stores/metadata.go

```diff
@@ -1160,7 +1168,54 @@ func (s *SQLStore) ObjectEntries(ctx context.Context, bucket, path, prefix, mark
 		utf8.RuneCountInString(path + prefix), // HAVING SUBSTR(name, 1, ?) = ? AND name != ?
 		path + prefix,                         // HAVING SUBSTR(name, 1, ?) = ? AND name != ?
 		path,                                  // HAVING SUBSTR(name, 1, ?) = ? AND name != ?
-	}, havingParams...), limit, offset)
+	}
+
+	// build marker expr
+	markerExpr := "1 = 1"
+	var markerParams []interface{}
+	if usingMarker {
+		switch sortBy {
+		case api.ObjectSortByHealth:
+			var markerHealth float64
+			if err = s.db.
+				Raw(fmt.Sprintf(`SELECT health FROM (%s ORDER BY name) as n WHERE name >= ? LIMIT 1`, objectsQuery), append(objectsQueryParams, marker)...).
```

Comment:
The marker is a very specific object id right? So it seems like it would be faster to run `objectHealth` here for a specific object instead of the massive listing query.

Reply:
Markers are specific indeed but I figured we can't rely on the fact that the marker is still there, that's why I perform the list query and do `WHERE name >= [marker] LIMIT 1`. I optimised the query though!